.github/workflows/main-workflow.yml

```diff
@@ -17,10 +17,10 @@ jobs:
         run:  sudo npm install
       
       - name: tshirts fail
-        run: bash .github/workflows/expectfail.sh "node tshirts.js"
+        run: "node tshirts.js"
 
       - name: misaligned fail
-        run: bash .github/workflows/expectfail.sh "node misaligned.js"
+        run: "node misaligned.js"
 
       - name: alerter fail
-        run: bash .github/workflows/expectfail.sh "node alerter.js"
+        run: "node alerter.js"
```

Comment:
You have separated the test code into alerter.spec.js, but the GitHub action is running alerter.js.
Switch it to run the alerter.spec.js. Same for the others.
Then you are sure of your functionality when the action passes (and shows a green tick)
You can see the run in the Actions tab above (like https://github.com/clean-code-personal/test-failer-in-js-adrian-numocity/actions/runs/6118242302/job/16606053976)

Reply:
Will do